### PR TITLE
Apply routine updates to fix PCT with more recent Jenkins versions and JDK11

### PR DESCRIPTION
**Pull request by Vlatombe**
* Update jenkins.version to 2.277.1
* Update matching bom
* Update maven wrapper to a current version
* Update wiremock as the previous version had guava incompatibilities with new core

Overlaps with #261 so maintainer will need to pick one of these (this PR being more conservative using an older Jenkins baseline, but doesn't cover the guava upgrade)

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by aHenryJard — APPROVED** [review]

**Review by kerogers-cloudbees — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java [hunk 1]
```diff
@@ -74,7 +75,7 @@ public void shouldGuessNameIfNotProvided() throws Exception {
         GitHubServerConfig input = new GitHubServerConfig("");
         input.setApiUrl(CUSTOM_GH_SERVER);
         assertThat(input.getName(), is(nullValue()));
-        assertThat(input.getDisplayName(), is("some (http://some.com)"));
```
Comment (KostyaSha) [inline src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java:77]: what's the reason for this change?

Comment (twasyl) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java:77]: I can't tell as it's working as well. @Vlatombe any reason behind this change?

Comment (Vlatombe) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java:77]: Looks like the test is failing with `mvn test -Dtest=GitHubServerConfigTest -Djenkins.version=2.332`. Looking up why but it looks related to guava upgrade.

Comment (Vlatombe) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java:77]: There is a swallowed error when running against jenkins 2.332

```
java.lang.NoSuchFieldError: EXACT
	at com.google.common.net.InternetDomainName.findSuffixOfType(InternetDomainName.java:171)
	at com.google.common.net.InternetDomainName.<init>(InternetDomainName.java:152)
	at com.google.common.net.InternetDomainName.from(InternetDomainName.java:210)
	at jenkins.scm.api.SCMName.resolveInternetDomainName(SCMName.java:143)
	at jenkins.scm.api.SCMName.innerFromUrl(SCMName.java:113)
	at jenkins.scm.api.SCMName.fromUrl(SCMName.java:67)
	at org.jenkinsci.plugins.github.config.GitHubServerConfigTest.given__url_with(GitHubServerConfigTest.java:83)
[...]
```

This is caused by the `wiremock:standalone` dependency which is vendoring `com.google.thirdparty.publicsuffix.PublicSuffixPatterns`

Comment (Vlatombe) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java:77]: The proper fix would be to remove the wiremock `standalone` classifier once the Jenkins baseline is raised to 2.332 or later. Until then, this test change seems to be the only way to run tests with both baselines (for PCT runs purpose against recent Jenkins baselines)

Comment (jtnord) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java:77]: You should always use the wiremock standalone due to the mass of library conflicts- however the version used is very old and may not be correctly packaged.

should use 
```
<dependency>
  <groupId>com.github.tomakehurst</groupId>
  <artifactId>wiremock-jre8-standalone</artifactId>
  <version>xxx</version>
</dependency>
```

Comment (Vlatombe) [reply src/test/java/org/jenkinsci/plugins/github/config/GitHubServerConfigTest.java]: I upgraded wiremock, looks good.

File: src/test/java/org/jenkinsci/plugins/github/test/GHMockRule.java [hunk 2]
```diff
@@ -13,6 +12,7 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import wiremock.org.eclipse.jetty.http.HttpStatus;
```
Comment (jtnord) [inline src/test/java/org/jenkinsci/plugins/github/test/GHMockRule.java:15]: NIT: generally I would not rely on on internal implementations, but use `java.net.HttpURLConnection.HTTP_CREATED`, but this was existing anyway.

**Review by jtnord — APPROVED** [review]

**Comment by KostyaSha** [comment]
hope squash merge is not a problem for everyone

**Comment by kerogers-cloudbees** [comment]
@KostyaSha Can you release this plugin? Or do you know who can so I can ask them?

**Comment by KostyaSha** [comment]
> @KostyaSha Can you release this plugin?

done
